### PR TITLE
LF: Comparisons fail at runtime if comparing local vs global CIDs

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR enforces that LF Comparisons (both Ordering and Equality)
fails at runtime if comparing local vs global CID.

This corresponds to the second task of #10504.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala [hunk 1]
```diff
@@ -106,27 +106,38 @@ object Ordering extends scala.math.Ordering[SValue] {
...
-          Bytes.ordering.compare(suffix1, suffix2)
-        else
-          // We crash here because we should have catch this beforehand
-          // when preprocessing or importing values.
-          throw SError.SErrorCrash(
-            getClass.getCanonicalName + ".compare",
-            "Unexcpected contract ID freshness Error",
-          )
+  private[this] def compareCid(cid1: ContractId, cid2: ContractId): Int = {
+    cid1 match {
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala:110]: Personally I found the pattern match on the pair we had before easier to follow but that’s clearly subjective so your choice

**Review by cocreature — APPROVED** [review]
Thanks!

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala]: 45ad0f16f278fad24faddd34a8445fb1bba3e009

**Comment by remyhaemmerle-da** [comment]
@moritzkiefer-da following the discussion we had this morning I revert the special case of create an exercise 45ad0f16f278fad24faddd34a8445fb1bba3e009

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala [hunk 2]
```diff
@@ -2751,6 +2764,13 @@ object EngineTest {
...
   }
 
+  private val dummySuffix = Bytes.assertFromString("00")
+
+  private def suffix(tx: Tx.Transaction) =
+    data.assertRight(tx.suffixCid(_ => dummySuffix))
+
+  // Mimics Canton reinterpreation
+  // requires a suffixed transaction.
   private def reinterpret(
```
Comment (remyhaemmerle-da) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:2774]: This function mimic Canton reinterpretion, for this reason I change the test that use it, to require CID suffixing.

**Pull request merged**